internal/controller/metrics.go

```diff
@@ -0,0 +1,35 @@
+package controller
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+const (
+	metricsNamespace = "nyallocator"
+)
+
+var (
+	CurrentNodesVec = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: metricsNamespace,
+		Name:      "current",
+		Help:      "Number of nodes under the NodeTemplate",
+	}, []string{"name"})
```

Comment:
Based on my experience, it is better to use the resource type as the label key, like the kube-state-metrics' metrics. 
If you use `name` label, you may need to relabel when combining with other components' metrics.

```suggestion
	}, []string{"nodetemplate"})
```